force-app/main/default/classes/ServiceScheduleSelector_TEST.cls

```diff
@@ -0,0 +1,82 @@
+/*
+ *
+ *  * Copyright (c) 2022, salesforce.com, inc.
+ *  * All rights reserved.
+ *  * SPDX-License-Identifier: BSD-3-Clause
+ *  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ *
+ */
+
+@IsTest
+public with sharing class ServiceScheduleSelector_TEST {
+    private static BasicStub permissionValidatorStub = new BasicStub(
+        PermissionValidator.class
+    );
+    private static ServiceScheduleSelector scheduleSelector = new ServiceScheduleSelector();
+
+    @TestSetup
+    private static void createScheduleData() {
+        TestDataFactory.generateServiceData();
+        TestDataFactory.generateServiceScheduleData('Daily', System.today());
+    }
+
+    @IsTest
+    private static void testgetServiceScheduleFromId() {
+        ServiceSchedule__c expectedSchedule = [
+            SELECT Id, Name
+            FROM ServiceSchedule__c
+            LIMIT 1
+        ];
+
+        Test.startTest();
+        ServiceSchedule__c actualSchedule = scheduleSelector.getServiceScheduleFromId(
+            expectedSchedule.Id
+        );
+        Test.stopTest();
+
+        System.assertEquals(
+            expectedSchedule.Id,
+            actualSchedule.Id,
+            'The Id of the returned record should match the record we requested'
+        );
+
+        System.assertEquals(
+            expectedSchedule.Name,
+            actualSchedule.Name,
+            'The Name of the returned record should match the record we requested'
+        );
+    }
+
+    @IsTest
+    private static void testGetServiceScheduleNoAccess() {
+        ServiceSchedule__c expectedSchedule = [SELECT Id FROM ServiceSchedule__c LIMIT 1];
```

Comment:
Super nitpicky so feel free to ignore me but I would maybe name this existingSchedule since we no longer expect it to be returned.